lib/backgroundscanner.php

```diff
@@ -42,89 +49,50 @@ public static function check() {
 			\OCP\Util::writeLog('files_antivirus', __METHOD__.', exception: '.$e->getMessage(), \OCP\Util::ERROR);
 			return;
 		}
-
-		$serverContainer = \OC::$server;
-		/** @var $serverContainer \OCP\IServerContainer */
-		$root = $serverContainer->getRootFolder();
-
-		// scan the found files
+	
+		$view = new \OC\Files\View('/');
 		while ($row = $result->fetchRow()) {
-			$file = $root->getById($row['fileid']); // this should always work ...
-			if (!empty($file)) {
-				$file = $file[0];
-				$storage = $file->getStorage();
-				$path = $file->getInternalPath();
-				self::scan($file->getId(), $path, $storage);
-			} else {
-				// ... but sometimes it doesn't, try to get the storage
-				$storage = self::getStorage($serverContainer, $row['id']);
-				if ($storage !== null && $storage->is_dir('')) {
-					self::scan($row['fileid'], $row['path'], $storage);
-				} else {
-					\OCP\Util::writeLog('files_antivirus', 'Can\'t get \OCP\Files\File for id "'.$row['fileid'].'"', \OCP\Util::ERROR);
-				}
+			$path = $view->getPath($row['fileid']);
+			if (!is_null($path)) {
+				$item = new Item($view, $path, $row['fileid']);
+				$scanner = new Scanner($this->appConfig);
+				$status = $scanner->scan($item);					
+				$status->dispatch($item, true);
 			}
 		}
+		\OC_Util::tearDownFS();
 	}
+	
+	/**
+	 * A hack to access files amd views. Better than before.
+	 */
+	protected function initFS(){
+		//get a random user (needed to mount FS)
```

Comment:
@DeepDiver1975 I need any existing user to init FS. Otherwise path won't be resolved.